libs/application/template-api-modules/src/lib/modules/templates/parental-leave/parental-leave.service.ts

```diff
@@ -34,12 +42,43 @@ export class ParentalLeaveService {
     )
   }
 
+  async getSelfEmployedPdf(application: Application) {
+    if (
+      (application.answers.employer as { isSelfEmployed: 'yes' | 'no' })
+        .isSelfEmployed === 'no'
+    ) {
+      return undefined
+    }
+
+    try {
+      const filename = (application.answers.employer as {
+        selfEmployed: { file: { key: string }[] }
+      }).selfEmployed.file?.[0].key
+      const Key = `${application.id}/${filename}`
+      const file = await this.s3
+        .getObject({ Bucket: this.attachmentBucket, Key })
+        .promise()
+      const fileContent = file.Body as Buffer
+
+      if (!fileContent) {
+        throw new Error('File content was undefined')
+      }
+
+      return fileContent.toString('base64')
+    } catch (e) {
+      this.logger.error('Cannot get self employed attachment', { e })
+      return undefined
+    }
+  }
+
   async sendApplication({ application }: TemplateApiModuleActionProps) {
     const nationalRegistryId = application.applicant
+    const attachment = await this.getSelfEmployedPdf(application)
```

Comment:
Shouldn't we move the self employment check outside of the function

```typescript
let attachment

if (selfEmployed) {
  attachment = await this.getSelfEmployedPdf(application)
}
```

And by doing that, we can change lines 68-70 to not fail silently:

```typescript
catch (e) {
  this.logger.error('Cannot get self employed attachment', { e })
  return undefined
}
```

Reply:
Sure that sounds good to me 👍

Reply:
However, we still need the try/catch around the S3 function, but I will throw an error instead